sup3r/preprocessing/data_handling/dual_data_handling.py

```diff
@@ -0,0 +1,614 @@
+            f'lr_handler.sample_shape {lr_handler.sample_shape} are '
+            f'incompatible. Must be {hr_shape} and {lr_shape}.'
+        )
+        assert lr_handler.sample_shape == lr_shape, msg
+
+        if hr_handler.data is not None and lr_handler.data is not None:
+            hr_shape = self.hr_data.shape
+            lr_shape = (
+                hr_shape[0] // self.s_enhance,
+                hr_shape[1] // self.s_enhance,
+                hr_shape[2] // self.t_enhance,
+                hr_shape[3],
+            )
+            msg = (
+                f'hr_data.shape {self.hr_data.shape} and '
+                f'lr_data.shape {self.lr_data.shape} are '
+                f'incompatible. Must be {hr_shape} and {lr_shape}.'
+            )
+            assert self.lr_data.shape == lr_shape, msg
+
+        if self.lr_val_data is not None and self.hr_val_data is not None:
+            hr_shape = self.hr_val_data.shape
+            lr_shape = (
+                hr_shape[0] // self.s_enhance,
+                hr_shape[1] // self.s_enhance,
+                hr_shape[2] // self.t_enhance,
+                hr_shape[3],
+            )
+            msg = (
+                f'hr_val_data.shape {self.hr_val_data.shape} '
+                f'and lr_val_data.shape {self.lr_val_data.shape}'
+                f' are incompatible. Must be {hr_shape} and {lr_shape}.'
+            )
+            assert self.lr_val_data.shape == lr_shape, msg
+
+    @property
+    def grid_mem(self):
+        """Get memory used by a feature at a single time step
+
+        Returns
+        -------
+        int
+            Number of bytes for a single feature array at a single time step
+        """
+        grid_mem = np.product(self.lr_grid_shape)
+        # assuming feature arrays are float32 (4 bytes)
+        return 4 * grid_mem
+
+    @property
+    def feature_mem(self):
+        """Number of bytes for a single feature array. Used to estimate
+        max_workers.
+
+        Returns
+        -------
+        int
+            Number of bytes for a single feature array
+        """
+        feature_mem = self.grid_mem * len(self.lr_time_index)
+        return feature_mem
+
+    @property
+    def sample_shape(self):
+        """Get lr sample shape"""
+        return self.lr_dh.sample_shape
+
+    @property
+    def lr_sample_shape(self):
+        """Get lr sample shape"""
+        return self.lr_dh.sample_shape
+
+    @property
+    def hr_sample_shape(self):
+        """Get hr sample shape"""
+        return self.hr_dh.sample_shape
+
+    @property
+    def features(self):
+        """Get list of features in each data handler"""
+        return self.hr_dh.features
+
+    @property
+    def lr_time_index(self):
+        """Return time index for low_res data"""
+        return self._lr_time_index
+
+    @lr_time_index.setter
+    def lr_time_index(self, time_index):
+        """Set time index for low_res data"""
+        self._lr_time_index = time_index
```

Comment:
There's no purpose in having a property and a setter if you're not including any additional logic or checks in either method. This is literally the same as exposing `self.lr_time_index` as a public unprotected attribute. You could argue it's nice to have the additional docstring but i think the variable names are self-explanatory and i dont think you can programmatically retrieve the docstring of a property/setter from a terminal anyway.

Reply:
Good point. These were cluttering things anyway.